TrainingExtensions/torch/src/python/aimet_torch/model_preparer.py

```diff
@@ -310,17 +303,12 @@ def prepare_model(model: torch.nn.Module, modules_to_exclude: List[torch.nn.Modu
      torch.fx.symbolic_trace API in detail.
     :return: Modified pytorch Model
     """
-    model.eval()
-    device = get_device(model)
-    symbolic_traced_model = _trace_model(model, modules_to_exclude, concrete_args)
+    with in_eval_mode(model):
+        traced_model = _trace_model(model, modules_to_exclude, concrete_args)
 
     # Prepare model and perform checks to make sure the graph is well-formed.
-    _prepare_helper(symbolic_traced_model)
-    _verify_symbolic_traced_model(symbolic_traced_model)
-
-    symbolic_traced_model.eval()
-    symbolic_traced_model.to(device)
-    return symbolic_traced_model
+    _prepare_traced_model(traced_model)
```

Comment:
Does prepare traced model also need to be done inside the with in_eval_mode context?